context.go

```diff
@@ -639,15 +644,19 @@ func (c *Context) parseFlag(flags []*Flag, match string) (err error) {
 	for _, flag := range flags {
 		long := "--" + flag.Name
 		short := "-" + string(flag.Short)
+		neg := "--no-" + flag.Name
 		candidates = append(candidates, long)
 		if flag.Short != 0 {
 			candidates = append(candidates, short)
 		}
-		if short != match && long != match {
+		if short != match && long != match && neg != match {
```

Comment:
This should be conditional on the tag being set, otherwise existing flags could break, I believe. eg. if a CLI defines `--no-foo` and `--foo`, this will currently change that logic if I'm not mistaken.

Reply:
I see. The problem I ran in to was that since we're trying to support a flag that technically doesn't exist, without this check here it'll enter the continue block and throw an error for using an un-supported flag. Any ideas? Should I just check if match already starts with `--no-`?

Reply:
I think it just needs to be `&& !(match == neg && flag.Tag.Negatable)`?

I think this case could use a test too, to make sure that existing CLIs continue to work. Thanks!